src/main/java/com/mentorship/vineservice/services/VineService.java

```diff
@@ -0,0 +1,9 @@
+package com.mentorship.vineservice.services;
+
+import com.mentorship.vineservice.domain.Vine;
+
+public interface VineService {
+
+    Long saveVineAndReturnId(Vine vine);
```

Comment:
I think we can name this method just "saveVine", and pay attention that we can receive as a method param not a domain object